src/SwaggerProvider.DesignTime/OperationCompiler.fs

```diff
@@ -137,54 +162,70 @@ type OperationCompiler (schema:SwaggerObject, defCompiler:DefinitionCompiler) =
+                | Some (BodyMultipart parts) ->
+                    <@ Http.RequestStream(%address,
                             httpMethod = restCall,
-                            headers = (%%heads : array<string*string>),
-                            query = (%%queries : (string * string) list),
-                            customizeHttpRequest = (%%customizeHttpRequest : Net.HttpWebRequest -> Net.HttpWebRequest)) @@>
-                | Some (FormData, b) ->
-                    <@@ Http.RequestString(%%address,
+                            body = HttpRequestBody.Multipart(string (Guid.NewGuid()), %parts),
+                            headers = %heads,
+                            query = %queries,
+                            customizeHttpRequest = %customizeHttpRequest) @>                              
+                | Some (BodyForm formData) ->
+                    <@ Http.RequestStream(%address,
                             httpMethod = restCall,
-                            headers = (%%heads : array<string*string>),
-                            body = HttpRequestBody.FormValues (%%b : seq<string * string>),
-                            query = (%%queries : (string * string) list),
-                            customizeHttpRequest = (%%customizeHttpRequest : Net.HttpWebRequest -> Net.HttpWebRequest)) @@>
-                | Some (Body, b)     ->
-                    <@@ let body = SwaggerProvider.Internal.RuntimeHelpers.serialize (%%b : obj)
-                        Http.RequestString(%%address,
+                            headers = %heads,
+                            body = HttpRequestBody.FormValues %formData,
+                            query = %queries,
+                            customizeHttpRequest = %customizeHttpRequest) @>
+                | Some (BodyObject b) ->
+                    <@  let body = SwaggerProvider.Internal.RuntimeHelpers.serialize %b
+                        Http.RequestStream(%address,
                             httpMethod = restCall,
-                            headers = (%%heads : array<string*string>),
+                            headers = %heads,
                             body = HttpRequestBody.TextRequest body,
-                            query = (%%queries : (string * string) list),
-                            customizeHttpRequest = (%%customizeHttpRequest : Net.HttpWebRequest -> Net.HttpWebRequest))
-                    @@>
-                | Some (x, _) -> failwith ("Payload should not be able to have type: " + string x)
-
+                            query = %queries,
+                            customizeHttpRequest = %customizeHttpRequest) @>
             // Return deserialized object
-            let value = <@@ SwaggerProvider.Internal.RuntimeHelpers.deserialize
-                                (%%result : string) retTy @@>
+
+            let value = 
+                <@@ 
+                    let stream = (%result).ResponseStream
+                    if isReturnFile 
```

Comment:
I don't like how I'm explicitly checking here.

I also don't like how I'm just returning a stream.  I think I'd like to see if we can default some kind of file name, perhaps from the Content-Disposition header of the response if it exists?